core/trino-main/src/main/java/io/trino/operator/exchange/ScaleWriterPartitioningExchanger.java

```diff
@@ -171,6 +171,6 @@ private long sendPageToPartition(Consumer<Page> buffer, Page pageSplit)
         long retainedSizeInBytes = pageSplit.getRetainedSizeInBytes();
         memoryManager.updateMemoryUsage(retainedSizeInBytes);
         buffer.accept(pageSplit);
-        return retainedSizeInBytes;
+        return pageSplit.getSizeInBytes();
```

Comment:
What's wrong about the existing code ?
We're calling `page.compact()` or `page.copyPositions` before `sendPageToPartition`, so what difference will this change make ?

Reply:
`getRetainedSizeInBytes` does not equal `getSizeInBytes` after `compact` or `copyPositions`. An example is dictionary. `getSizeInBytes` was always used to measure flow of data

Reply:
If this is just about measuring data processed, then why don't we call `getSizeInBytes` once on the full Page before splitting it rather than calling it on each partitioned Page and then adding that up.

Reply:
You are right, https://github.com/trinodb/trino/pull/21668